File: clojure/p12.clj
(ns p12
  (:use [euler :only (factors naturals find-first timed-test)]))

;; From Project Euler solution pdf
(defn count-nth-tri-num-factors [n]
  (let [D (fn [x] (count (factors x)))]
    (if (even? n)
      (* (D (/ n 2)) (D (inc n)))
      (* (D n) (D (/ (inc n) 2))))))

(defn nth-tri-num [n]
  (* n (/ (inc n) 2)))

(timed-test
 76576500
 (let [n (find-first #(> (count-nth-tri-num-factors %) 500) (naturals))]
   (nth-tri-num n)))

File: cpp/p12.cpp
#include "euler.h"

using namespace std;

int main() {
  int n = 1;

  while( get_divisors( get_s_gonal( n, 3 ) ).size() <= 500 )
    n++;
	
  cout << get_s_gonal( n, 3 ) << endl;
}

